### PR TITLE
Don't fail on warnings truncation on the addon validation page (bug 1139674)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1139674](https://bugzilla.mozilla.org/show_bug.cgi?id=1139674)

### CODE REVIEW

**Comment by magopian** [comment]
Thanks @mstriemer for the very thorough investigation in the bug :heart: 

I know I didn't need to fix in both places, but thought it would be better "just in case". What do you think?

File: static/js/common/upload-addon.js [hunk 1]
```diff
@@ -341,7 +341,7 @@
                             ], current, matched, messages = [],
                             // this.id is in the form ["testcases_javascript_instanceactions", "_call_expression", "createelement_variable"],
                             // we usually only match one of the elements.
-                            matchId = function (id) {return _.contains(this.id, id);};
+                            matchId = function (id) {return (this.hasOwnProperty('id') && _.contains(this.id, id));};
```
Comment (mstriemer) [inline static/js/common/upload-addon.js]: I don't think the parens are needed here. Might be nice to break the `return` onto its own line now that this is a bit longer too.

File: apps/devhub/utils.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def limit_validation_results(validation, is_compatibility=False):
             validation['messages'].append({
                 'tier': 1,
                 'type': msgtype,
+                'id': [],  # To respect the message structure, see bug 1139674.
```
Comment (mstriemer) [inline apps/devhub/utils.py]: Mind giving this an id? I think there are normally 3 strings but I'm not sure if that's required, or if it just ends up that way. I'd try to stretch it to 3. `['validation', 'messages', 'truncated']` or something, which is terrible but matches more closely with the other `id`s.

**Comment by mstriemer** [comment]
I'm all for being thorough. I'd do both fixes.

r+wc

**Comment by magopian** [comment]
Thanks @mstriemer for the review, i've modified the PR, is that still ok with you?

**Comment by mstriemer** [comment]
:+1: lgtm

**Pull request merged**